internal/resource/resource.go

```diff
@@ -39,6 +51,50 @@ func (r *Resource) Parse() (*unstructured.Unstructured, error) {
 	return u, u.UnmarshalJSON([]byte(r.Manifest.Manifest))
 }
 
+func (r *Resource) NeedsToBePatched(current *unstructured.Unstructured) bool {
+	if r.Patch == nil || current == nil {
+		return false
+	}
+
+	curjson, err := current.MarshalJSON()
+	if err != nil {
+		return false
+	}
+
+	patchedjson, err := r.Patch.Apply(curjson)
+	if err != nil {
+		return false
+	}
+
+	patched := &unstructured.Unstructured{}
```

Comment:
How about renaming this function to `GetPatched` and return both a bool and the patched resource to avoid recomputing it?

Reply:
IIRC (I wrote this on a Friday lol) this is the only place we apply the patch locally. The second application is when we send the raw patch json to apiserver.